File: common/phaser/messages.go
package phaser

import (
	"time"
)

type ScanCompletedMessage struct {
	ReportID string `json:"report_id"`
	File     File   `json:"file"`
}

type File struct {
	Path   string `json:"path"`
	MD5    string `json:"md5,omitempty"`
	SHA1   string `json:"sha1,omitempty"`
	SHA256 string `json:"sha256,omitempty"`
	SHA512 string `json:"sha512,omitempty"`
}

type ScanQueuedMessage struct {
	ScanID   string   `json:"scan_id"`
	Targets  []string `json:"targets"`
	Profile  string   `json:"profile"`
	ReportID string   `json:"report_id"`
}

type ScanStartedMessage struct {
	ReportID  string    `json:"report_id"`
	StartedAt time.Time `json:"started_at"`
}

File: src/worker/messages.rs
use serde::{Serialize, Deserialize};

// Message is used to send and receive messages between services
// kernel -> phaser
#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(tag = "type", content = "data")]
pub enum In {
    #[serde(rename = "scan_queued")]
    ScanQueued(ScanQueued),
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ApiResponse {
    pub error: Option<String>,
    pub data: Option<ApiData>,
    pub status: u32,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(tag = "type", content = "data")]
pub enum ApiData {
    #[serde(rename = "scan_queued")]
    ScanQueued(ScanQueued),
}


#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ScanQueued {
    pub scan_id: uuid::Uuid,
    pub targets: Vec<String>,
    pub profile: String,
    pub report_id: uuid::Uuid
}

// MessageOut is used to send and receive messages between services
// phaser -> kernel
#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(tag = "type", content = "data")]
pub enum Out {
    #[serde(rename = "scan_completed")]
    ScanCompleted(ScanCompleted),
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ScanStarted {
    pub report_id: uuid::Uuid,
    pub started_at: String,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ScanCompleted {
    pub report_id: String,
    // pub sah512: S,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct File {
    pub path: String,
    pub md5: Option<String>,
    pub sha1: Option<String>,
    pub sha256: Option<String>,
    pub sha512: Option<String>,
}
